tests/utils/test_transfer.py

```diff
@@ -0,0 +1,29 @@
+import metatensor.torch
+import torch
+from metatensor.torch import Labels, TensorMap
+from metatensor.torch.atomistic import System
+
+from metatrain.utils.transfer import systems_and_targets_to_dtype_and_device
+
+
+def test_systems_and_targets_to_dtype_and_device():
+    system = System(
+        positions=torch.tensor([[1.0, 1.0, 1.0]]),
+        cell=torch.tensor([[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0]]),
+        types=torch.tensor([1]),
+    )
+    targets = TensorMap(
+        keys=Labels.single(),
+        blocks=[metatensor.torch.block_from_array(torch.tensor([[1.0]]))],
+    )
+
+    systems, targets = systems_and_targets_to_dtype_and_device(
+        [system], {"energy": targets}, torch.float64, torch.device("meta")
+    )
+
+    assert systems[0].positions.dtype == torch.float64
```

Comment:
could you add an assert that everything was using another dtype/device before the move?